File: src/boj/binarysearch/BOJ10816.java
package boj.binarysearch;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ10816 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[20000001];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            int card = Integer.parseInt(st.nextToken());
            arr[card + 10000000]++;
        }
        n = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < n; i++) {
            int card = Integer.parseInt(st.nextToken());
            sb.append(arr[card + 10000000]).append(" ");
        }
        System.out.println(sb);
    }
}

File: sample/ps_kotlin/boj/src/main/kotlin/boj10816.kt
import java.util.*

fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.out.bufferedWriter()

    val n = br.readLine().toInt()
    val st1 = StringTokenizer(br.readLine(), " ")
    val cards = IntArray(20_000_001)
    repeat(n) { cards[st1.nextToken().toInt() + 10_000_000]++ }

    val m = br.readLine().toInt()
    val st2 = StringTokenizer(br.readLine(), " ")
    repeat(m) {
        bw.write("${cards[st2.nextToken().toInt() + 10_000_000]} ")
    }
    bw.flush()
}
